common/mails/courses/index.ts

```diff
@@ -234,40 +235,53 @@ const isPromotionValid = (publishedAt: Date, capacity: number, alreadyPromoted:
     return capacity < 0.75 && alreadyPromoted === false && (daysDiff === null || daysDiff > 3);
 };
 
-export async function sendPupilCourseSuggestion(course: Course | Prisma.course, subcourse: Prisma.subcourse, actionId: ActionID) {
-    // TODO: Reenable once we stabilized course promotions
-    return;
+export async function sendPupilCoursePromotion(subcourse: Prisma.subcourse) {
+    const predictedPupilResponseRate = 5; // in percent
 
-    const minGrade = subcourse.minGrade;
-    const maxGrade = subcourse.maxGrade;
     const courseCapacity = await getCourseCapacity(subcourse);
     const { alreadyPromoted, publishedAt } = subcourse;
-    const courseSubject = course.subject;
-    const grades = [];
+    if (!isPromotionValid(publishedAt, courseCapacity, alreadyPromoted)) {
+        throw new Error(`Promotion for Subcourse(${subcourse.id}) is not valid!`);
+    }
 
+    const course = await getCourse(subcourse.courseId);
+    const minGrade = subcourse.minGrade;
+    const maxGrade = subcourse.maxGrade;
+    const grades = [];
     for (let grade = minGrade; grade <= maxGrade; grade++) {
         grades.push(`${grade}. Klasse`);
     }
 
-    const pupils = await prisma.pupil.findMany({
+    let pupils = await prisma.pupil.findMany({
         where: { active: true, isParticipant: true, grade: { in: grades } },
     });
 
-    const context = await getNotificationContextForSubcourse(course, subcourse);
-
-    if (actionId === 'available_places_on_subcourse' && !isPromotionValid(publishedAt, courseCapacity, alreadyPromoted)) {
-        throw new Error(`Cannot send course promotion for Subcourse(${subcourse.id})`);
-    }
-
-    let sentCount = 0;
-    for (let pupil of pupils) {
+    const courseSubject = course.subject;
+    const filteredPupils = pupils.filter(async (pupil) => {
```

Comment:
A Promise is always truthy, the filter won't work. Maybe we can create a Set of pupil ids of the Course participants, then do `.filter(pupil => !participantIds.has(pupil.id))`

Reply:
Alternatively we could also filter on the database.